pkg/generator/generator.go

```diff
@@ -351,52 +415,159 @@ func (g *Generator) renderPkg() error {
 		return err
 	}
 
 	buf = &bytes.Buffer{}
-	if err := renderAPITypesFile(buf, kind, version); err != nil {
+	if err := renderGenericFile(buf, "apis/<apiDirName>/<version>/types.go", apiTypesTmpl, tmplData{
+		Kind:    kind,
+		Version: version,
+	}); err != nil {
 		return err
 	}
 	return writeFileAndPrint(filepath.Join(apiDir, types), buf.Bytes(), defaultFileMode)
 }
 
 func renderStubFiles(stubDir, repoPath, kind, apiDirName, version string) error {
 	buf := &bytes.Buffer{}
-	if err := renderHandlerFile(buf, repoPath, kind, apiDirName, version); err != nil {
+	if err := renderGenericFile(buf, "stub/handler.go", handlerTmpl, tmplData{
+		OperatorSDKImport: sdkImport,
+		RepoPath:          repoPath,
+		Kind:              kind,
+		APIDirName:        apiDirName,
+		Version:           version,
+	}); err != nil {
 		return err
 	}
 	return writeFileAndPrint(filepath.Join(stubDir, handler), buf.Bytes(), defaultFileMode)
 }
 
+type tmplData struct {
+	VersionNumber string
+
+	OperatorSDKImport string
+	StubImport        string
+	K8sutilImport     string
+	SDKVersionImport  string
+
+	APIVersion string
+	Kind       string
+
+	RepoPath   string
+	APIDirName string
+	Version    string
+
+	ProjectName string
+	GroupName   string
+
+	KindSingular string
+	KindPlural   string
+
+	Image string
+	Name  string
+
+	PackageName string
+	ChannelName string
+	CurrentCSV  string
+
+	CRDVersion     string
+	CSVName        string
+	CatalogVersion string
+}
+
+func (g *Generator) generateDirStructure() error {
+	if err := os.MkdirAll(g.projectName, defaultDirFileMode); err != nil {
+		return err
+	}
+
+	cpDir := filepath.Join(g.projectName, cmdDir, g.projectName)
+	if err := os.MkdirAll(cpDir, defaultDirFileMode); err != nil {
+		return err
+	}
+
+	cp := filepath.Join(g.projectName, configDir)
+	if err := os.MkdirAll(cp, defaultDirFileMode); err != nil {
+		return err
+	}
+
+	dp := filepath.Join(g.projectName, deployDir)
+	if err := os.MkdirAll(dp, defaultDirFileMode); err != nil {
+		return err
+	}
+
+	op := filepath.Join(g.projectName, olmCatalogDir)
+	if err := os.MkdirAll(op, defaultDirFileMode); err != nil {
+		return err
+	}
+
+	bDir := filepath.Join(g.projectName, buildDir)
+	if err := os.MkdirAll(bDir, defaultDirFileMode); err != nil {
+		return err
+	}
+	cDir := filepath.Join(g.projectName, codegenDir)
+	if err := os.MkdirAll(cDir, defaultDirFileMode); err != nil {
+		return err
+	}
+
+	if err := os.MkdirAll(filepath.Join(g.projectName, versionDir), defaultDirFileMode); err != nil {
+		return err
+	}
+
+	v := version(g.apiVersion)
+	adn := apiDirName(g.apiVersion)
+	apiDir := filepath.Join(g.projectName, apisDir, adn, v)
+	if err := os.MkdirAll(apiDir, defaultDirFileMode); err != nil {
+		return err
+	}
+	sDir := filepath.Join(g.projectName, stubDir)
+	if err := os.MkdirAll(sDir, defaultDirFileMode); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func renderGenericFile(w io.Writer, fileLoc string, fileTmpl string, info tmplData) error {
```

Comment:
Could you rename this to `renderFile` and add a comment above it describing what it does?